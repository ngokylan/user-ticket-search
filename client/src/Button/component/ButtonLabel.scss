@import '../../_lib/style/index';

.btn {
  opacity: 1;

  &__label {
    font-family: $font-family;
    text-align: center;
    position: relative;
    z-index: 1;
    @include text-style-body-rg;
  }

  &--is-uppercase {
    .btn__label {
      @include text-style-button;
    }
  }
}
