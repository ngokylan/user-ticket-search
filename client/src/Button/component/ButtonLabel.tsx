import React from 'react';
import { ButtonLabel as ButtonLabelType, ButtonTypes } from '../type';
import './ButtonLabel.scss';

export type ButtonLabelProps = {
  children?: ButtonLabelType;
  id?: string;
};

function ButtonLabel({ children, id }: ButtonLabelProps) {
  return (
    <span
      className="btn__label"
      data-testid={`btn-label-${id || 'default'}`}
    >
      {children}
    </span>
  );
}

export default ButtonLabel;
