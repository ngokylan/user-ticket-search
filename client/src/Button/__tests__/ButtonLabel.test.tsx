import React from 'react';
import { mount } from 'enzyme';
import ButtonLabel, { ButtonLabelProps } from '../component/ButtonLabel';

// Expected default props.
const defaultProps: ButtonLabelProps = undefined;
const mountComponent = (customProps: any = {}) => {
  const props = { ...defaultProps, ...customProps };
  const wrapper = mount(<ButtonLabel {...props} />);
  const component = wrapper.find('.btn__label');

  return { props, wrapper, component };
};

describe('Button::ButtonLabel', () => {
  describe('rendering', () => {
    it('should render component without any crash', () => {
      const { wrapper } = mountComponent();
      expect(wrapper).toHaveLength(1);
    });

    it.each([[{}], [{ id: 'custom-id', icon: 'custom-icon' }]])(
      'should be match with snapshot',
      props => {
        const { wrapper } = mountComponent(props);
        expect(wrapper).toMatchSnapshot();
      }
    );
  });

  describe('props', () => {
    describe('default', () => {
      it('should have expected default props', () => {
        expect(JSON.stringify(defaultProps)).toBe(
          JSON.stringify(ButtonLabel.defaultProps)
        );
      });

      it.each([
        ['btn-label-default', {}],
        ['btn-label-custom-id', { id: 'custom-id' }],
      ])('should render test ID properly', (expectedTestId, props) => {
        const { component } = mountComponent(props);
        expect(component.prop('data-testid')).toBe(expectedTestId);
      });
    });

    describe('children', () => {
      it('should render children', () => {
        const { wrapper } = mountComponent({ children: 'children' });
        expect(wrapper.find('span').text()).toBe('children');
      });
    });
  });
});
