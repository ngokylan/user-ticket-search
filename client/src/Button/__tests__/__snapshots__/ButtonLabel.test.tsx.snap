// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button::ButtonLabel rendering should be match with snapshot 1`] = `
<span
  className="btn__label"
  data-testid="btn-label-default"
/>
`;

exports[`Button::ButtonLabel rendering should be match with snapshot 2`] = `
<span
  className="btn__label"
  data-testid="btn-label-custom-id"
/>
`;
